language/evm/examples/sources/Table.move

```diff
@@ -0,0 +1,35 @@
+/// This module defines Table for EVM.
+module Evm::Table {
+    native struct Table<K, V> has store, copy, drop;
+
+    /// Create an empty Table.
+    native public fun empty<K, V>(): Table<K, V>;
+
+    /// Acquire an immutable reference to the value which `key` maps to.
+    /// Aborts if there is no entry for `key`.
+    native public fun borrow<K, V>(table: &Table<K, V>, key: &K): &V;
+
+    /// Acquire a mutable reference to the value which `key` maps to.
+    /// Aborts if there is no entry for `key`.
+    native public fun borrow_mut<K, V>(table: &mut Table<K, V>, key: &K): &mut V;
```

Comment:
This may in fact need to extend the prover's borrow analysis and perhaps also Move compiler. The borrow_mut on vectors is hard wired in those, as it creates a borrow edge labelled by the vector index; something similar is needed here. Not that this is a blocker for this, but just need to keep this additional complexity in mind.

Reply:
OK. I will check this with Todd later.